airflow/providers/sftp/operators/sftp_batch.py

```diff
@@ -0,0 +1,312 @@
+            put_dir_txt_files = SFTPOperator(
+                task_id="put_dir_txt_files",
+                ssh_conn_id="ssh_default",
+                local_folder="/tmp/dir_for_local_transfer/",
+                remote_folder="/tmp/dir_for_remote_transfer/",
+                regexp_mask=".*[.]txt",
+                operation=SFTPOperation.PUT,
+                create_intermediate_dirs=True
+            )
+    Move `/tmp/file1.txt` file
+        from local to remote folder `/tmp/dir_for_remote_transfer/`
+            put_files = SFTPOperator(
+                task_id="put_dir_txt_files",
+                ssh_conn_id="ssh_default",
+                local_files_path=["/tmp/file1.txt",],
+                remote_folder="/tmp/dir_for_remote_transfer/",
+                operation=SFTPOperation.PUT,
+                create_intermediate_dirs=True
+            )
+    Move all files
+        from remote folder `/tmp/dir_for_remote_transfer/` to local folder `/tmp/dir_for_local_transfer/`
+            get_dir = SFTPOperator(
+                task_id="put_dir_txt_files",
+                ssh_conn_id="ssh_default",
+                local_folder="/tmp/dir_for_local_transfer/",
+                remote_folder="/tmp/dir_for_remote_transfer/",
+                operation=SFTPOperation.GET,
+                create_intermediate_dirs=True
+            )
+    Move `/tmp/file1.txt` file
+        from remote to local folder `/tmp/dir_for_local_transfer/`
+            get_files = SFTPOperator(
+                task_id="put_dir_txt_files",
+                ssh_conn_id="ssh_default",
+                local_folder="/tmp/dir_for_local_transfer/",
+                remote_files_path=["/tmp/file1.txt",],
+                operation=SFTPOperation.GET,
+                create_intermediate_dirs=True
+            )
+    """
+
+    template_fields = (
+        'remote_host',
+        'local_files_path',
+        'remote_files_path',
+        'local_folder',
+        'remote_folder',
+        'regexp_mask',
+    )
+
+    def __init__(
+        self,
+        *,
+        ssh_hook=None,
+        ssh_conn_id=None,
+        remote_host=None,
+        local_files_path: Optional[List[str]] = None,
+        remote_files_path: Optional[List[str]] = None,
+        local_folder: Optional[str] = None,
+        remote_folder: Optional[str] = None,
+        regexp_mask=None,
+        operation=SFTPOperation.PUT,
+        confirm=True,
+        create_intermediate_dirs=False,
+        force=False,
+        list_as_str=False,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.ssh_hook = ssh_hook
+        self.ssh_conn_id = ssh_conn_id
+        self.remote_host = remote_host
+        self.local_files_path = local_files_path
+        self.remote_files_path = remote_files_path
+        self.local_folder = local_folder
+        self.remote_folder = remote_folder
+        self.regexp_mask = regexp_mask
+        self.operation = operation
```

Comment:
```suggestion
        self.operation = operation.lower()
```

There is a little bit of a mixed bag of using `self.operation.lower()` and `self.operation` in some of the validations. When the value is not lowercase some of the validations won't work as expected. If you set the value to be lower case initially you don't need to repeat applying the function throughout.